ports/nrf/common-hal/bleio/Adapter.c

```diff
@@ -39,17 +39,34 @@
 #include "supervisor/usb.h"
 #include "shared-bindings/bleio/Adapter.h"
 #include "shared-bindings/bleio/Address.h"
+#include "mpconfigboard.h" // for BOARD_HAS_CRYSTAL
 
 STATIC void softdevice_assert_handler(uint32_t id, uint32_t pc, uint32_t info) {
     mp_raise_msg_varg(&mp_type_AssertionError,
                       translate("Soft device assert, id: 0x%08lX, pc: 0x%08lX"), id, pc);
 }
 
+static inline bool board_has_crystal(void) {
+#ifdef BOARD_HAS_CRYSTAL
+    return BOARD_HAS_CRYSTAL == 1;
```

Comment:
I would have written 'return BOARD_HAS_CRYSTAL;', so that any nonzero value is interpreted as TRUE.

Reply:
Also, it's probably a bug if BOARD_HAS_CRYSTAL is *not* defined for some boards, so maybe just use BOARD_HAS_CRYSTAL directly, not via `#ifdef`

Reply:
This is actually copied from the SAMD port. I thought it would be best to follow existing code.
I can change this and i think the same change should be done for the SAMD port.